mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -244,4 +283,21 @@ static void overlayTableProperties(Configuration configuration, TableDesc tableD
     // this is an exception to the interface documentation, but it's a safe operation to add this property
     props.put(InputFormatConfig.TABLE_SCHEMA, schemaJson);
   }
+
+  private static class NonSerializingConfig implements Serializable {
+
+    private transient volatile Configuration conf;
```

Comment:
Do we set the config from multiple threads?
Do we set the config at all?

I think this could be `final` instead of `volatile`, as we so not set the conf, we create a new `NonSerializingConfig`

Reply:
Yep, good point